vars/abapEnvironmentPullGitRepo.groovy

```diff
@@ -1,292 +1,61 @@
 import com.sap.piper.Utils
-import com.sap.piper.BashUtils
-import groovy.json.JsonSlurper
 import hudson.AbortException
 import groovy.transform.Field
-import java.util.UUID
-import java.util.regex.*
 
 @Field def STEP_NAME = getClass().getName()
-@Field Set STEP_CONFIG_KEYS = [
-    /**
-     * Specifies the host address of the SAP Cloud Platform ABAP Environment system
-     */
-    'host',
-    /**
-     * Jenkins CredentialsId containing the communication user and password of the communciation scenario SAP_COM_0510
-     */
-    'credentialsId',
-    /**
-     * Specifies the name of the Repository (Software Component) on the SAP Cloud Platform ABAP Environment system
-     */
-    'repositoryName',
-    'cloudFoundry',
-        /**
-         * Cloud Foundry API endpoint.
-         * @parentConfigKey cloudFoundry
-         */
-        'apiEndpoint',
-        'credentialsId',
-        /**
-         * Cloud Foundry target organization.
-         * @parentConfigKey cloudFoundry
-         */
-        'org',
-        /**
-         * Cloud Foundry target space.
-         * @parentConfigKey cloudFoundry
-         */
-        'space',
-        /**
-         * Cloud Foundry service instance, for which the service key will be created.
-         * @parentConfigKey cloudFoundry
-         */
-        'serviceInstance',
-        /**
-         * Cloud Foundry service key, which will be created.
-         * @parentConfigKey cloudFoundry
-         */
-        'serviceKey',
-    /** @see dockerExecute */
-    'dockerImage',
-    /** @see dockerExecute */
-    'dockerWorkspace'
-]
-@Field Set GENERAL_CONFIG_KEYS = STEP_CONFIG_KEYS
-@Field Set PARAMETER_KEYS = STEP_CONFIG_KEYS
-@Field Map CONFIG_KEY_COMPATIBILITY = [cloudFoundry: [apiEndpoint: 'cfApiEndpoint', credentialsId: 'cfCredentialsId', org: 'cfOrg', space: 'cfSpace', serviceInstance: 'cfServiceInstance', serviceKey: 'cfServiceKey']]
-/**
- * Pulls a Repository (Software Component) to a SAP Cloud Platform ABAP Environment system.
- *
- * This is either possible by providing the host and the credentialsId of the communication arrangement or by providing access to a service key for the communication arrangement SAP_COM_0510 on cloud foundry.
- *
- * !!! note "Git Repository and Software Component"
- *       In SAP Cloud Platform ABAP Environment Git repositories are wrapped in Software Components (which are managed in the App "Manage Software Components")
- *       Currently, those two names are used synonymous.
- */
-@GenerateDocumentation
+@Field String METADATA_FILE = 'metadata/abapEnvironmentPullGitRepo.yaml'
+
 void call(Map parameters = [:]) {
     handlePipelineStepErrors(stepName: STEP_NAME, stepParameters: parameters, failOnError: true) {
 
         def script = checkScript(this, parameters) ?: this
-        Map configuration = ConfigurationHelper.newInstance(this)
+
+        Set configKeys = ['dockerImage', 'dockerWorkspace']
+        Map jenkinsConfig = ConfigurationHelper.newInstance(this)
```

Comment:
config resolution in groovy no more required.

This is done by the two `piper` calls below, one for the `--context` (e.g. secrets and Docker stuff), the other one for the step configuration